lib/zendesk_apps_support/validations/requirements.rb

```diff
@@ -56,10 +58,22 @@ def missing_required_fields(requirements)
         end
 
         def excessive_requirements(requirements)
-          count = requirements.values.map(&:values).flatten.size
+          count = requirements.values.map do |req|
+            req.is_a?(Hash) ? req.values : req
+          end.flatten.size
           ValidationError.new(:excessive_requirements, max: MAX_REQUIREMENTS, count: count) if count > MAX_REQUIREMENTS
         end
 
+        def excessive_custom_objects_requirements(requirements)
+          custom_objects = requirements['custom_objects']
+          return if custom_objects.nil?
```

Comment:
Hello 👋 

How do you feel about changing `return if custom_objects.nil?` to `return unless custom_objects`, for consistency with line 111 in this file.

Also, I am curious what happens if someone includes the extreme edge case value `false` here,
i.e `requirements['custom_objects'] = false`. But perhaps this is not important. What do you think?

Reply:
Yep good idea, will update the code.

The edge case where `requirements['custom_objects'] = false` will skip over the validation for `custom_objects` which is fine since none are defined. The build will then fail and error in ZAM as a hash is expected to be returned. 
For this case we could handle the semantically incorrect values in ZAM or add validations to check that the requirements data structure contains the correct types. I'll add a comment to the custom objects implementation PR and we should follow this up in another story if we think its necessary

Reply:
Awesome. Thank you.